File: haskell/p002.hs
slice :: Int -> Int -> [a] -> [a]
slice from to xs = take (to - from + 1) (drop from xs)
fibs = 1 : 1 : zipWith (+) fibs (tail fibs)
fib n = fibs !! n
solution = print(sum [x | x <- slice 0 1000 fibs, even x && x < 4000000])

File: rust/src/p002.rs
pub fn solution(){
    let mut fibs: [i32; 1000] = [0; 1000];
    let mut index = 2;
    let mut sum = 2;
    fibs[0] = 1;
    fibs[1] = 2;

    loop {
        fibs[index] = fibs[index-1] + fibs[index-2];
        if fibs[index] >= 4000000 {
            break;
        }
        else if fibs[index] % 2 == 0 {
            sum += fibs[index];
        }
        index+=1;
    }
    println!("{}", sum); 
}